### PR TITLE
explode backbone router

### PR DESCRIPTION
**Pull request by nicolasThal**

### CODE REVIEW

File: BackofficeBundle/Resources/public/coffee/backboneRouter.coffee [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  generateUrl: (routeName, paramsObject) ->
+    optionalParam = /\(([^)]*):([^)]*)\)/g
+    namedParam = /():([^/]*)/g
+    route = @routePatterns[routeName]
+
+    replaceFunction = ->
+      key = arguments[2]
+      if paramsObject[key] then arguments[1] + paramsObject[key] else ''
+
+    if typeof route != 'undefined'
```
Comment (itkg-alavieille) [inline BackofficeBundle/Resources/public/coffee/backboneRouter.coffee]: you can replace this line by ``if route?``

File: BackofficeBundle/Resources/public/coffee/backboneRouter.coffee [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      route = route.replace(optionalParam, replaceFunction).replace(namedParam, replaceFunction)
+    else
+      alert 'Error, route name is unknown'
+      return false
+    return route
+  addParametersToRoute: (options) ->
+    Router = this
+    fragment = Backbone.history.fragment
+    routes = _.pairs(Router.routePatterns)
+    route = null
```
Comment (itkg-alavieille) [inline BackofficeBundle/Resources/public/coffee/backboneRouter.coffee]: i think declarations : 
``
route = null 
matched = undefined
paramsObject = null
paramsKeys = null
``
are useless with coffee

File: BackofficeBundle/Resources/public/coffee/backboneRouter.coffee [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    return
+  showHome: ->
+    @navigate 'dashboard', true
+    return
+  addRoutePattern: (routeName, routePattern) ->
+    @routePatterns[routeName] = routePattern
+    return
+  generateRoutePatterns: ->
+    currentRouter = this
+    $.each @routes, (routePattern, routeName) ->
```
Comment (itkg-alavieille) [inline BackofficeBundle/Resources/public/coffee/backboneRouter.coffee]: you can replace this line by

**Comment by nicolasThal** [comment]
@itkg-alavieille pr updated

File: BackofficeBundle/Resources/public/coffee/backboneRouter.coffee [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  generateUrl: (routeName, paramsObject) ->
+    optionalParam = /\(([^)]*):([^)]*)\)/g
+    namedParam = /():([^/]*)/g
+    route = @routePatterns[routeName]
+
+    replaceFunction = ->
+      key = arguments[2]
+      if paramsObject[key] then arguments[1] + paramsObject[key] else ''
+
+    if route
```
Comment (itkg-alavieille) [inline BackofficeBundle/Resources/public/coffee/backboneRouter.coffee]: you forgot the character ``?``
if route => if route?

**Comment by itkg-alavieille** [comment]
:+1:  after the fix

**Pull request merged**